nordicsemi/dfu/dfu_transport_mesh.py

```diff
@@ -320,10 +310,12 @@ def send_firmware(self, firmware):
             time.sleep(self.interval)
 
         # Wait for any final missing packet requests
-        time.sleep(DfuTransportMesh.DATA_REQ_WAIT_TIME)
-        while len(self.requested_packets) > 0:
-            self.send_packet(SerialPacket(self.requested_packets.popleft()))
-            time.sleep(self.interval)
+        timeout = time.time() + DfuTransportMesh.DATA_REQ_WAIT_TIME
+        while time.time() <= timeout:
+            while len(self.requested_packets) > 0:
+                self.send_packet(SerialPacket(self.requested_packets.popleft()))
+                time.sleep(self.interval)
+                timeout = time.time() + DfuTransportMesh.DATA_REQ_WAIT_TIME
```

Comment:
Is it infinite loop while `len(self.requested_packets) > 0`? Is it correct?

Reply:
`self.requested_packets.popleft()` pops the packet from `requested_packets`.